marin/post_training/environments/aqua_rat_env.py

```diff
@@ -0,0 +1,60 @@
+import logging
+from typing import ClassVar
+
+import datasets
+from tqdm.auto import tqdm
+
+from .math_env import MathEnv
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+
+class AquaRatEnv(MathEnv):
+    """
+    AQUA-RAT (Algebra Question Answering with Rationales) is a large (~100K) crowdsourced collection of
+    multiple-choice algebraic/math word problems where each item includes the problem statement,
+    five answer options (A-E), a natural-language step-by-step rationale explaining the solution,
+    and the correct option.
+
+    HuggingFace: https://huggingface.co/datasets/deepmind/aqua_rat
+
+    @article{ling2017program,
+      title={Program induction by rationale generation: Learning to solve and explain algebraic word problems},
+      author={Ling, Wang and Yogatama, Dani and Dyer, Chris and Blunsom, Phil},
+      journal={ACL},
+      year={2017}
+    }
+    """
+
+    HF_DATASET_NAME: ClassVar[str] = "deepmind/aqua_rat"
+
+    INSTRUCTION: str = (
+        "Show your work in <think> </think> tags. And return just the letter of the correct answer "
+        "(one of A, B, C, D or E) as the final answer in <answer> </answer> "
```

Comment:
I think multiple choice is bad for RL since you can just get spurious rationales that happen to get the answer correct.  Can we convert this into exact match on answers?